src/api/index.tsx

```diff
@@ -266,3 +276,50 @@ export const getDiff = async ({
     query: path ? { file: path } : undefined,
   });
 };
+
+export const createOrUpdateComment = async ({
+  /* istanbul ignore next */
+  _callApi = callApi,
+  addonId,
+  apiState,
+  cannedResponseId,
+  comment,
+  commentId,
+  fileName,
+  line,
+  versionId,
+}: {
+  _callApi?: typeof callApi;
+  addonId: number;
+  apiState: ApiState;
+  cannedResponseId: number | void;
+  comment: string | void;
+  commentId: number | void;
+  fileName: string | null;
+  line: number | null;
+  versionId: number;
+}) => {
+  if (cannedResponseId === undefined && comment === undefined) {
+    throw new Error('Either cannedResponseId or comment must be specified');
```

Comment:
How could you do it in TS?

Reply:
Let me see if I can come up with something...

Reply:
I believe that something like this would work:

```diff
diff --git a/src/api/index.tsx b/src/api/index.tsx
index 6e470c3..56453e3 100644
--- a/src/api/index.tsx
+++ b/src/api/index.tsx
@@ -277,6 +277,28 @@ export const getDiff = async ({
   });
 };

+type BaseCreateOrUpdateCommentParams = {
+  _callApi?: typeof callApi;
+  addonId: number;
+  apiState: ApiState;
+  commentId: number | void;
+  fileName: string | null;
+  line: number | null;
+  versionId: number;
+};
+
+type WithCannedResponseParams = {
+  cannedResponseId: number;
+  comment: void;
+};
+type WithCommentParams = {
+  cannedResponseId: void;
+  comment: string;
+};
+
+type CreateOrUpdateCommentParams = BaseCreateOrUpdateCommentParams &
+  (WithCannedResponseParams | WithCommentParams);
+
 export const createOrUpdateComment = async ({
   /* istanbul ignore next */
   _callApi = callApi,
@@ -288,24 +310,7 @@ export const createOrUpdateComment = async ({
   fileName,
   line,
   versionId,
-}: {
-  _callApi?: typeof callApi;
-  addonId: number;
-  apiState: ApiState;
-  cannedResponseId: number | void;
-  comment: string | void;
-  commentId: number | void;
-  fileName: string | null;
-  line: number | null;
-  versionId: number;
-}) => {
-  if (cannedResponseId === undefined && comment === undefined) {
-    throw new Error('Either cannedResponseId or comment must be specified');
-  }
-  if (cannedResponseId && comment) {
-    throw new Error('cannedResponseId and comment cannot both be specified');
-  }
-
+}: CreateOrUpdateCommentParams) => {
   let endpoint = `reviewers/addon/${addonId}/versions/${versionId}/draft_comments`;
   if (commentId) {
     endpoint = `${endpoint}/${commentId}`;
```